src/main/java/Entities/Item.java

```diff
@@ -1,6 +1,31 @@
 package main.java.Entities;
 
-public class Item {
+import main.java.EntityInterfaces.ItemImpl;
+import main.java.Entities.Ingredient;
+
+public class Item implements ItemImpl {
     private Ingredient ingredient;
     private int quantity;
+    private Boolean optional = false;
+
+    public Item(Ingredient ingredient, int quantity) {
+        this.ingredient = ingredient;
+        this.quantity = quantity;
+    }
+
+    @Override
+    public Ingredient ingredient() {
+        return this.ingredient;
+    }
+
+    @Override
+    public int quantity() {
+        return this.quantity;
+    }
+
+    @Override
+    public Boolean optional(Boolean value) {
+        this.optional = value;
+        return this.optional;
+    }
```

Comment:
This is not supposed to be a setter function, it should take no args and simply return the value of optional.